### PR TITLE
Rename system chaining to system welding

### PR DESCRIPTION
**Pull request by alice-i-cecile**
System chaining sounds very innocuous and helpful: new users will often want to use it for either general purpose system communication or system ordering.

Both of these are very wrong. Renaming it to system welding better captures the irreversible one-to-one nature of the operation.

Docs have also been cleaned up slightly to steer users in the correct direction.

### CODE REVIEW

File: examples/ecs/system_welding.rs [hunk 1]
```diff
@@ -1,23 +1,29 @@
+/// System welding is a valuable but niche tool that allows you to pass the output from one system
+/// as input into a second system on a one-to-one basis.
+///
+/// It is most useful for error handling, or applying similar system adaptors to existing code.
+/// For more general-purpose system communication, you should probably use Events instead.
+/// For system ordering, use `.before` and `.after`.
 use anyhow::Result;
 use bevy::prelude::*;
 
 fn main() {
```
Comment (MinerSebas) [inline examples/ecs/system_welding.rs:10]: All other examples (with the exception of the removal-detection one) with comments that are placed before `main`, have those after the Imports.
For consistency, I would do the same here.

```suggestion
use anyhow::Result;
use bevy::prelude::*;

/// System welding is a valuable but niche tool that allows you to pass the output from one system
/// as input into a second system on a one-to-one basis.
///
/// It is most useful for error handling, or applying similar system adaptors to existing code.
/// For more general-purpose system communication, you should probably use Events instead.
/// For system ordering, use `.before` and `.after`.
fn main() {
```

**Review by MinerSebas — APPROVED** [review]

**Comment by mockersf** [comment]
I quite dislike `weld`, but I'm not sure I have anything better to offer. Maybe `and_then` to at least follow something from rust std lib. `fuse` could be closer to the `weld` idea.

**Comment by alice-i-cecile** [comment]
That's perfectly fair: `weld` was Ratys's suggestion on Discord when I asked.

`pipe` is probably my personal favorite, but I'm not sure if the usage here is close enough to the standard use. `fuse` feels a bit off to me (as does `weld`) due to the lack of implied directionality to it.

**Comment by andresovela** [comment]
What about `join` or `combine` or `merge`? They sound like irreversible words and they are not as obscure as `weld`.

Other options:
https://www.thesaurus.com/browse/link
https://www.thesaurus.com/browse/combine
https://www.thesaurus.com/browse/fuse

I think `pipe` has a connotation similar to `chain` that will make users want to use it for system ordering.

**Comment by alice-i-cecile** [comment]
> I think pipe has a connotation similar to chain that will make users want to use it for system ordering.

I think I agree with this.

I'd like to avoid `join`, `combine` and `merge` due to overloading if possible.

The thesaurus links were helpful though: `splice` captures what we want: a directed, irreversible, one-to-one joining of two systems.

**Comment by TheRawMeatball** [comment]
I'm personally not a big fan of splice tbh, since it isn't a very commonly used word, making the API trickier for non-native speakers. I think we should stick to more commonly used words. I'd personally be happier with `fuse` or `weld`.

**Comment by andresovela** [comment]
I agree, as a non-native speaker, I'd have to look the word up in the dictionary to double-check the meaning. I'd be more inclined to `fuse` than `weld`.

**Comment by cart** [comment]
Theres also a lot of overlap with `map` semantics. Basically identical if you consider system arguments to fill the same role as captured variables in closures.

**Comment by alice-i-cecile** [comment]
Closing this out, without consensus and a much better name it's not worth the churn right now.

**Pull request closed without merging**